src/GafferImage/ImageReader.cpp

```diff
@@ -356,8 +367,11 @@ ImageReader::DefaultColorSpaceFunction &ImageReader::defaultColorSpaceFunction()
 void ImageReader::affects( const Plug *input, AffectedPlugsContainer &outputs ) const
 {
 	ImageNode::affects( input, outputs );
-
-	if( input == intermediateMetadataPlug() || input == colorSpacePlug() )
+	if( input == fileNamePlug() || input == refreshCountPlug() || input == channelInterpretationPlug() )
```

Comment:
We don't pull on any of these plugs in the compute for `fileValid`. Other than the frame range plugs (that are dealt with below) we just pull on `__oiioReader.fileValid`. Technically we're not allowed to pull on that, because it's not actually an input to our node - Gaffer can only track dependencies between inputs/outputs on the same node, and along connections. I suspect the additional plugs here are masking that problem, but I wouldn't want to bet that they will always be sufficient.

I think the technically correct thing to do is :

- Add a new `__intermediateFileValid` _input_ to ImageReader, which we connect `__oiioReader.fileValid` into. This mirrors the similar plugs we use for receiving the image and metadata from the internal nodes.
- Declare the dependency on `__intermediateFileValid` here, and pull on that in the hash/compute in place of `__oiioReader.fileValid.